src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/LibuvConnection.cs

```diff
@@ -128,70 +128,57 @@ private static void ReadCallback(UvStreamHandle handle, int status, object state
 
         private void OnRead(UvStreamHandle handle, int status)
         {
-            var normalRead = status >= 0;
-            var normalDone = status == LibuvConstants.EOF;
-            var errorDone = !(normalDone || normalRead);
-            var readCount = normalRead ? status : 0;
-
-            if (normalRead)
+            if (status >= 0)
             {
-                Log.ConnectionRead(ConnectionId, readCount);
-            }
-            else
-            {
-                _socket.ReadStop();
+                Log.ConnectionRead(ConnectionId, status);
 
-                if (normalDone)
+                Debug.Assert(_currentWritableBuffer != null);
+                var currentWritableBuffer = _currentWritableBuffer.Value;
+                currentWritableBuffer.Advance(status);
+                var flushTask = currentWritableBuffer.FlushAsync();
+
+                if (!flushTask.IsCompleted)
                 {
-                    Log.ConnectionReadFin(ConnectionId);
+                    // We wrote too many bytes too the reader so pause reading and resume when
```

Comment:
Since you're moving this, might as well fix the typo: the second "too" should be "to" 😄